File: Arrays and Strings/isUnique/Unique.java
import java.util.*;

class Unique {

    public boolean isUnique(String str) {
	HashSet<Character> mem = new HashSet<Character>();
	for (int i = 0; i < str.length(); ++i)
	    if (!mem.add(str.charAt(i))) return false;
	return true;
    }

    public static void main(String[] args) {
	Scanner sc = new Scanner(System.in);
	System.out.print("Enter a string: ");
	String str = sc.next();
	Unique unq = new Unique();
	
	if (unq.isUnique(str))
	    System.out.println("The string is unique");
	else
	    System.out.println("the string is not unique");
    }
}

File: 01 Arrays and Strings/Arrays and Strings/1.1-Is-Unique/unique.cpp
#include <iostream>
#include <string>
#include <unordered_set>

using namespace std;

template<typename T>
bool containsKey(unordered_set<T> m, T key) {
  return m.find(key) != m.end();
}

bool isUnique(string str) {
  unordered_set<char> mem; // Hash set for "remembering" characters
  for (char c : str) {
    // Check if the character has already been seen
    if (containsKey(mem, c)) return false;
    // "Remember" the character
    mem.insert(c);
  }
  return true;
}

int main() {
  string s;
  cout << "Enter a string: " << endl;
  cin >> s;
  cout << (isUnique(s) ? "The string is unique" : "The string is not unique") << endl;
  return 0;
}
